src/server/scripts/Zandalar/AtalDazar/boss_priestess_alun_za.cpp

```diff
@@ -0,0 +1,684 @@
+        events.ScheduleEvent(EVENT_CHECK_ENERGY, 500ms);
+
+        if (IsHeroicOrHigher() || IsMythicPlus())
+        {
+            events.ScheduleEvent(EVENT_SPIRIT_OF_GOLD, 9100ms);
+            events.ScheduleEvent(EVENT_CORRUPTED_GOLD, 3s);
+        }
+    }
+
+    void JustDied(Unit* /*killer*/) override
+    {
+        _JustDied();
+        instance->SendEncounterUnit(ENCOUNTER_FRAME_DISENGAGE, me);
+
+        Talk(SAY_DIED);
+    }
+
+    void ExecuteEvent(uint32 eventId) override
+    {
+        switch (eventId)
+        {
+            case EVENT_GILDED_CLAWS:
+            {
+                Talk(SAY_GILDED_CLAWS_WARNING, me);
+                Talk(SAY_GILDED_CLAWS);
+                DoCastSelf(SPELL_GILDED_CLAWS);
+                events.Repeat(33500ms);
+                break;
+            }
+            case EVENT_TAINTED_BLOOD:
+            {
+                DoCast(SPELL_TAINTED_BLOOD_SELECTOR);
+                events.Repeat(2s, 4s);
+                break;
+            }
+            case EVENT_MOLTEN_GOLD:
+            {
+                DoCast(SPELL_AGITATE);
+                events.Repeat(16500ms);
+                break;
+            }
+            case EVENT_CHECK_ENERGY:
+            {
+                if (me->GetPower(POWER_ENERGY) >= 100)
+                {
+                    Talk(SAY_TRANSFUSION_WARNING);
+                    Talk(SAY_TRANSFUSION);
+                    DoCast(SPELL_TRANSFUSION);
+                }
+                events.Repeat(500ms);
+                break;
+            }
+            case EVENT_CORRUPTED_GOLD:
+            {
+                me->SummonCreature(NPC_CORRUPTED_GOLD, CorruptedGoldSpawnPosition[0], TEMPSUMMON_MANUAL_DESPAWN);
+                me->SummonCreature(NPC_CORRUPTED_GOLD, CorruptedGoldSpawnPosition[1], TEMPSUMMON_MANUAL_DESPAWN);
+                events.Repeat(3s);
+                break;
+            }
+            case EVENT_SPIRIT_OF_GOLD:
+            {
+                DoCast(SPELL_SPIRIT_OF_GOLD);
+                events.Repeat(35s);
+                break;
+            }
+            default:
+                break;
+        }
+    }
+
+    void UpdateAI(uint32 diff) override
+    {
+        if (!UpdateVictim())
+            return;
+
+        events.Update(diff);
+
+        if (me->HasUnitState(UNIT_STATE_CASTING))
+            return;
+
+        while (uint32 eventId = events.ExecuteEvent())
+        {
+            ExecuteEvent(eventId);
+            if (me->HasUnitState(UNIT_STATE_CASTING))
+                return;
+        }
+    }
+};
+
+// 130738 - Corrupted Gold
+struct npc_priestess_alun_za_corrupted_gold : public ScriptedAI
+{
+    npc_priestess_alun_za_corrupted_gold(Creature* creature) : ScriptedAI(creature) { }
+
+    void JustAppeared() override
+    {
+        DoCastSelf(SPELL_CORRUPTED_GOLD);
+
+        // TODO: orbs have collision but dont work properly so they are despawning when reach fixed position
```

Comment:
collision you mean when hitting the WMO to bounce back?

Reply:
yes but some of them get collisions with the cauldron and dont come out

Reply:
well, those points you're using as destination are not really using MovePositionToFirstCollision, that would fix that issue you're experiencing